src/audio/detect_test.c

```diff
@@ -430,74 +370,26 @@ static int test_keyword_get_config(struct comp_dev *dev,
 
 	trace_keyword("test_keyword_get_config()");
 
-	/* Copy back to user space */
-	bs = cd->config.size;
-	trace_value(bs);
-
-	if (bs == 0 || bs > size)
+	if (!cd->config)
 		return -EINVAL;
 
-	assert(!memcpy_s(cdata->data->data, size, &cd->config, bs));
-	cdata->data->abi = SOF_ABI_VERSION;
-	cdata->data->size = bs;
-
-	return ret;
-}
-
-static int test_keyword_get_model(struct comp_dev *dev,
-				  struct sof_ipc_ctrl_data *cdata, int size)
-{
-	struct comp_data *cd = comp_get_drvdata(dev);
-	size_t bs;
-	uint32_t crc;
-	int ret = 0;
-
-	trace_keyword("test_keyword_get_model()");
-
-	/* Copy back to user space */
-	if (cd->load_memory) {
-		bs = sizeof(uint32_t);
-
-		if (bs > size) {
-			trace_keyword_error("test_keyword_get_model() error: "
-					    "invalid size %d", bs);
-			return -EINVAL;
-		}
-
-		crc = crc32(cd->load_memory, cd->config.load_memory_size);
-
-		trace_keyword("test_keyword_get_model() crc: 0x%X", crc);
+	/* Copy back to user space, without the model */
+	bs = sizeof(struct sof_detect_test_config);
```

Comment:
bs = cd->config->size; ?

Reply:
size contains the model

Reply:
@slawblauciak we need to get this data(model) and send back to host driver side to dumping to end  user, so the size should be the whole size.

Reply:
The mailbox is not large enough for the entire model, it can't be copied there. And I can't exactly split it into chunks.